### PR TITLE
password_randomizer function

### PR DESCRIPTION
**Pull request by shibo911**
FIXES #5 

added a password randomizer_function. the function takes a single argument which is the required password length.  the function creates strong passwords by using the system's randomness through /dev/urandom.
Screenshot 2024-03-15 012537

### CODE REVIEW

**Comment by umeshSinghVerma** [comment]
Hii @shibo911 kindly create a new branch with for your feature and also improve PR message

File: scripts/password-generator [hunk 1]
```diff
@@ -1,5 +1,10 @@
...
-# Description:
-# Create a function that generates passwords, The password should consist of uppercase letters, lowercase letters, numbers, and special characters. The function should take in a single argument, which is the length of the password to generate. The function should return the generated password.
+password_randomizer() {
+    local length=$1
+    local password=$(cat /dev/urandom | tr -dc 'A-Za-z0-9!@#$%^&*()' | fold -w $length | head -n 1)
+    echo $password
+}
+
+
+generate_password $1
```
Comment (rootCircle) [inline scripts/password-generator:10]: did you meant `password_randomizer` here?

File: scripts/password-generator [hunk 2]
```diff
@@ -1,5 +1,10 @@
-#!/usr/bin/env bash
+#!/bin/bash
 
-# Issue: Implement Password Generation Functions
-# Description:
-# Create a function that generates passwords, The password should consist of uppercase letters, lowercase letters, numbers, and special characters. The function should take in a single argument, which is the length of the password to generate. The function should return the generated password.
+password_randomizer() {
+    local length=$1
+    local password=$(cat /dev/urandom | tr -dc 'A-Za-z0-9!@#$%^&*()' | fold -w $length | head -n 1)
```
Comment (rootCircle) [inline scripts/password-generator:5]: add error checking if args is empty

image

**Review by rootCircle — CHANGES_REQUESTED** [review]

**Comment by rootCircle** [comment]
@shibo911 great work! please go through the reviews right here and see to mark the relevant changes! ping me when done

**Comment by shibo911** [comment]
> @shibo911 great work! please go through the reviews right here and see to mark the relevant changes! ping me when done

@rootCircle i have made commits again and did the changes as you requested. please re-review it. thanks!

**Review by rootCircle — APPROVED** [review]

**Comment by rootCircle** [comment]
LGTM :-) Also, one peculiar thing, the `password-generator` does not has executable permission by default! Change that by `chmod` and then commit and push.

ping me here when you're done with those, so we can move forward with your PR!

**Comment by rootCircle** [comment]
Great going @shibo911! You did a great job! Thanks for your contribution! Accepted :tada:

**Pull request closed without merging**